File: ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/python/reading_data/sample10.py
import dv_processing as dv

# Open any camera
reader = dv.io.MonoCameraRecording("path/to/file.aedat4")

# Run the loop while camera is still connected
while reader.isRunning():
    # Read batch of events
    events = reader.getNextEventBatch()
    if events is not None:
        # Print received packet time range
        print(f"{events}")

File: ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/c++/reading_data/sample10.cpp
#include <dv-processing/io/mono_camera_recording.hpp>

int main() {
	// Open a file
	dv::io::MonoCameraRecording reader("path/to/file.aedat4");

	// Run the loop while data is available
	while (reader.isRunning()) {
		// Read batch of events, check whether received data is correct.
		if (const auto events = reader.getNextEventBatch(); events.has_value()) {
			// Print received event packet information
			std::cout << *events << std::endl;
		}
	}

	return 0;
}
